device-posture/device-posture.https.html

```diff
@@ -0,0 +1,89 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+//<script src="../resources/orientation-event-helpers.js"></script>
+<script>
+'use strict';
+
+async function cleanup() {
+  await test_driver.set_device_posture('continuous');
+}
+
+promise_test(async (t) => {
+  t.add_cleanup(cleanup);
+  const promise = new Promise(resolve => {
+    navigator.devicePosture.addEventListener("change", () => {
+      resolve();
+    });
+  });
+
+  const postures = ['folded', 'continuous', 'folded'];
+  for (let posture of postures) {
+    await test_driver.set_device_posture(posture);
+    await promise;
```

Comment:
You can't `await` for the same promise multiple times. After it resolves once, you're just getting the same result it gave you the first time.

Reply:
Because of request of using EventWatcher , this part of code changed and no promise is anymore used.